parser.y

```diff
@@ -6006,6 +6008,14 @@ SumExpr:
 	{
 		$$ = &ast.AggregateFuncExpr{F: $1, Args: []ast.ExprNode{$4}, Distinct: $3.(bool)}
 	}
+|	"JSON_OBJECTAGG" '(' BuggyDefaultFalseDistinctOpt Expression ',' BuggyDefaultFalseDistinctOpt Expression ')' OptWindowingClause
```

Comment:
```suggestion
|	"JSON_OBJECTAGG" '(' Expression ',' Expression ')' OptWindowingClause
```

MySQL's [`json_objectagg`](https://dev.mysql.com/doc/refman/8.0/en/group-by-functions.html#function_json-objectagg) does not support DISTINCT. And indeed, adding DISTINCT to the key is useless, since JSON objects would remove duplicated keys anyway.

(Perhaps add a test case to ensure `JSON_OBJECTAGG(DISTINCT a, b)` and `JSON_OBJECTAGG(a, DISTINCT b)` are parse failure)

Reply:
Thanks for your suggestions.I have tried it and json_objectagg does's support distinct but support all, so I add ALL to parser.y and add some test in parser_test.go as your suggestion.